### ISSUE TITLE
complex template string cause 'Maximum call stack size exceeded' error

### ISSUE THREAD

**Issue by clark-t** [babel/babel-eslint#636]
## env
node: 8.11.1
eslint: 4.19.1
babel-eslint: 8.2.3

## description

the code is:

```javascript
console.log(`it is ${
  'complex'.split('').map(str => {
    return `${str}`
  }).join('')
}`)
```
and causes this error:

```
Maximum call stack size exceeded
RangeError: Maximum call stack size exceeded
    at RegExp.test (<anonymous>)
    at OffsetStorage.getDesiredIndent (/Users/test/node_modules/eslint/lib/rules/indent.js:445:32)
    at OffsetStorage.getDesiredIndent (/Users/test/node_modules/eslint/lib/rules/indent.js:451:45)
    at OffsetStorage.getDesiredIndent (/Users/test/node_modules/eslint/lib/rules/indent.js:451:45)
    at OffsetStorage.getDesiredIndent (/Users/test/node_modules/eslint/lib/rules/indent.js:451:45)
    at OffsetStorage.getDesiredIndent (/Users/test/node_modules/eslint/lib/rules/indent.js:451:45)
    at OffsetStorage.getDesiredIndent (/Users/test/node_modules/eslint/lib/rules/indent.js:451:45)
    at OffsetStorage.getDesiredIndent (/Users/test/node_modules/eslint/lib/rules/indent.js:451:45)
    at OffsetStorage.getDesiredIndent (/Users/test/node_modules/eslint/lib/rules/indent.js:451:45)
    at OffsetStorage.getDesiredIndent (/Users/test/node_modules/eslint/lib/rules/indent.js:451:45)
```
my .eslintrc is:

```json
{
  "parser": "babel-eslint",
  "rules": {
      "indent": ["error", 2]
  }
}
```

when remove babel-eslint and change eslintrc to:
```json
{
  "parserOptions": {
    "ecmaVersion": 8
  },
  "rules": {
      "indent": ["error", 2]
  }
}
```
eslint runs well, so I think this bug should be caused by babel-eslint

**Comment by existentialism** [comment]
@clark-t thanks for the issue! This looks like a duplicate of https://github.com/babel/babel-eslint/issues/603, which was fixed by https://github.com/babel/babel-eslint/pull/610. We’ll cut a new release ASAP!

**Comment by existentialism** [comment]
@clark-t just published 8.2.4, which should fix the issue!
